cmake/stm32/common.cmake

```diff
@@ -33,16 +39,40 @@ if(NOT STM32_TOOLCHAIN_PATH)
     file(TO_CMAKE_PATH "${STM32_TOOLCHAIN_PATH}" STM32_TOOLCHAIN_PATH)
 endif()
 
-if(NOT STM32_TARGET_TRIPLET)
-    set(STM32_TARGET_TRIPLET "arm-none-eabi")
-    message(STATUS "No STM32_TARGET_TRIPLET specified, using default: " ${STM32_TARGET_TRIPLET})
-endif()
+function(stm32_autoset_toolchain_bin_path STM32_TARGET_TRIPLET RESULT_VAL_TOOLCHAIN_BIN_PATH)
+# set the OS utility to find a program path
+    if(MINGW OR CYGWIN OR WIN32)
+        set(OS_UTIL_SEARCH_CMD where)
+    elseif(UNIX OR APPLE)
+        set(OS_UTIL_SEARCH_CMD which)
+    endif()

```

Comment:
Why don't you use cmake's  [find_program](https://cmake.org/cmake/help/v3.17/command/find_program.html) ?

Reply:
But can I use the result of find_program inside an execute_process block?

Moreover initially my idea was to use the shell for *nix systems. The proper way of finding a program is through the use of the builtin shell function `command -v` (on some barebone systems like certain busybox builds, `which` is not installed). However this would result in a non standard way of calling the search command for windows/Linux because you should call `sh -c command -v <here the program to find` and in windows you don't do that. So I opted to do it like this for now. If there is a better way we can change it.